### PR TITLE
Fix #5465 by catching no-image exception

### LINKED ISSUE #5465: Building container fails after removing old image.
I found this issue while working on a fix for issue #4337.

By running the following sequence of commands on [this example repository](https://github.com/twelve17/dc-test):

```
git checkout tags/test2_run1
docker-compose build
docker rmi --force dctest_service1
docker-compose build
docker-compose up
```

Compose fails because the old image is never properly rebuilt from the old container:

```
WARNING: The Docker Engine you're using is running in swarm mode.

Compose does not use swarm mode to deploy services to multiple nodes in a swarm. All containers will be scheduled on the current node.

To deploy your application across the swarm, use `docker stack deploy`.

Recreating dctest_service1_1 ... error

ERROR: for dctest_service1_1  no such image: sha256:bb41503c2e723478308998afc7df0886fac0811621b70d61b28ca36418a41bb7: No such image: sha256:bb41503c2e723478308998afc7df0886fac0811621b70d61b28ca36418a41bb7

ERROR: for service1  no such image: sha256:bb41503c2e723478308998afc7df0886fac0811621b70d61b28ca36418a41bb7: No such image: sha256:bb41503c2e723478308998afc7df0886fac0811621b70d61b28ca36418a41bb7
ERROR: Encountered errors while bringing up the project.
```

### PR DESCRIPTION
**Pull request by Dahca**
Fix #5465 by catching a no-image exception in get_container_data_volumes. Added a case to the unit test to check that an empty existing image doesn't result in an exception.

### CODE REVIEW

**Comment by shin-** [comment]
The current error message is definitely not helpful, but I'm not sure we should be ignoring this exception, as it might mean loss of data for the user. Maybe we should have a prompt asking them if they want to proceed, or error out with a message recommending to backup the volume and run `down`?

**Comment by Dahca** [comment]
That seems reasonable. I've added a prompt that asks the user what they want to do on error. Is this the sort of prompt you had in mind?

**Comment by Dahca** [comment]
@shin- Do you have any more comments on this fix?

File: compose/service.py [hunk 1]
```diff
@@ -24,6 +24,7 @@
 from . import __version__
 from . import const
 from . import progress_stream
+from .cli.utils import yesno
```
[the remark this answers is not in the document]
Comment (shin-) [reply compose/service.py]: We should let the exception bubble up, handle it at the CLI level (with the prompt), then retry the operation with a `rebuild` flag or similar.

File: compose/service.py [hunk 2]
```diff
@@ -1283,12 +1284,36 @@ def merge_volume_bindings(volumes, tmpfs, previous_container, mounts):
...
+    try:
+        image_volumes = [
+            VolumeSpec.parse(volume)
+            for volume in
+            container.image_config['ContainerConfig'].get('Volumes') or {}
+        ]
+        return image_volumes
+    except ImageNotFound:
+        prompt = \
+            "WARNING: Image not found. Continue by rebuilding the image? [yN]"
```
Comment (shin-) [inline compose/service.py]: Ideally we would be a little more verbose here, to warn the user about the risk of data loss and recommend they back up  if necessary. The current message doesn't convey why someone wouldn't want to rebuild.

**Pull request closed without merging**